research/SemaphoreCircuitIntegration.md

```diff
@@ -0,0 +1,63 @@
+### Semaphore Circuit Overview  
+
+The Semaphore circuit is a key part of the Semaphore protocol, providing the cryptographic foundation for private and anonymous transactions. This document highlights the main challenges and opportunities in integrating the Semaphore circuit with the Stellar blockchain.  
+
+Semaphore uses components like **identity generation**, **Merkle proof verification**, and **nullifier creation** to ensure privacy and security in its transactions. However, adapting these features to Stellar involves addressing several compatibility and optimization challenges.  
+
+One of the main hurdles is that Semaphore relies on the BLS12-381 curve for cryptography, which doesn’t align directly with Stellar’s structure. Stellar’s unique ledger system and transaction model also add complexity to this integration.  
```

Comment:
Modify this so that it aligns with the new changes made and the research, which is that Stellar now supports the curve, however the Semaphore circuit uses another curve.

Reply:
Noted